packages/@dcl/ecs/src/engine/component.ts

```diff
@@ -191,18 +194,46 @@ export type ComponentDefinition<T extends ISchema = ISchema<any>> = {
   isDirty(entity: Entity): boolean
 }
 
-export function defineComponent<T extends ISchema>(
+export function defineComponent<
+  T extends ISchema,
+  ConstructorType = ComponentType<T>
+>(
   componentId: number,
-  spec: T
+  spec: T,
+  constructorDefault?: ConstructorType
   // meta: { syncFlags }
-): ComponentDefinition<T> {
+): ComponentDefinition<T, ConstructorType> {
   const data = new Map<Entity, ComponentType<T>>()
   const dirtyIterator = new Set<Entity>()
 
+  const defaultBuffer = createByteBuffer()
+  if (constructorDefault) {
+    spec.serialize(constructorDefault, defaultBuffer)
+  }
+
+  function getDefaultValue() {
+    if (constructorDefault) {
+      return spec.deserialize(
+        createByteBuffer({
+          writing: {
+            buffer: defaultBuffer.buffer(),
+            currentOffset: defaultBuffer.currentWriteOffset()
+          }
+        })
+      )
```

Comment:
I'd add a comment here to evaluate the performance of this method in the future. It seems to me that generating the defaults using this method instead of a deep-clone may be a lot of heavylifting. After feature parity we should revisit this and measure the performance

Reply:
https://github.com/decentraland/sdk/issues/430